mrtt-ui/src/library/useInitializeMonitoringForm.js

```diff
@@ -0,0 +1,39 @@
+import axios from 'axios'
+import { toast } from 'react-toastify'
+import { useEffect } from 'react'
+
+import formatApiAnswersForForm from './formatApiAnswersForForm'
+import language from '../language'
+
+const useInitializeMonitoringForm = ({
+  apiUrl,
+  questionMapping,
+  resetForm,
+  setIsLoading,
+  isEditMode = false
+}) => {
+  useEffect(
+    function initializeFormWithApiData() {
+      if (resetForm && apiUrl && isEditMode) {
+        setIsLoading(true)
+        axios
+          .get(apiUrl)
+          .then((response) => {
+            setIsLoading(false)
+            const initialValuesForForm = formatApiAnswersForForm({
+              apiAnswers: response.data.answers,
```

Comment:
main diff is there is no callback, we check for `isEditMode` and were getting answers from response.data.answers instead of response.data

Reply:
I see - is this only for monitoring sections?

Reply:
yup